File: src/domain/aiModel/entities/ModelConfig.ts
/**
 * 模型配置实体
 * 存储模型的基本配置信息
 */
export class ModelConfig {
  /** 模型唯一标识 */
  modelId: string;

  /** 模型显示名称 */
  modelName: string;

  /** 模型类型：image | chat */
  modelType: 'image' | 'chat';

  /** API 基础 URL */
  baseUrl: string;

  /** API Key */
  apiKey: string;

  /** 完整的 API URL（可选） */
  apiUrl?: string;

  /** 是否启用 */
  enabled: boolean;

  /** 创建时间 */
  createdAt: Date;

  /** 更新时间 */
  updatedAt: Date;

  constructor(data: {
    modelId: string;
    modelName: string;
    modelType: 'image' | 'chat';
    baseUrl: string;
    apiKey: string;
    apiUrl?: string;
    enabled?: boolean;
    createdAt?: Date;
    updatedAt?: Date;
  }) {
    this.modelId = data.modelId;
    this.modelName = data.modelName;
    this.modelType = data.modelType;
    this.baseUrl = data.baseUrl;
    this.apiKey = data.apiKey;
    this.apiUrl = data.apiUrl;
    this.enabled = data.enabled ?? true;
    this.createdAt = data.createdAt || new Date();
    this.updatedAt = data.updatedAt || new Date();
  }

  /**
   * 更新配置
   */
  update(data: Partial<{
    modelName: string;
    baseUrl: string;
    apiKey: string;
    apiUrl: string;
    enabled: boolean;
  }>): void {
    if (data.modelName !== undefined) this.modelName = data.modelName;
    if (data.baseUrl !== undefined) this.baseUrl = data.baseUrl;
    if (data.apiKey !== undefined) this.apiKey = data.apiKey;
    if (data.apiUrl !== undefined) this.apiUrl = data.apiUrl;
    if (data.enabled !== undefined) this.enabled = data.enabled;
    this.updatedAt = new Date();
  }
}

File: src-tauri/src/domain/model_config.rs
use serde::{Deserialize, Serialize};
use chrono::{DateTime, Utc};

/// 模型配置实体
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ModelConfig {
    /// 模型唯一标识
    pub model_id: String,

    /// 模型显示名称
    pub model_name: String,

    /// 模型类型：image | chat
    pub model_type: ModelType,

    /// API 基础 URL
    pub base_url: String,

    /// API Key
    pub api_key: String,

    /// 完整的 API URL（可选）
    pub api_url: Option<String>,

    /// 是否启用
    pub enabled: bool,

    /// 创建时间
    pub created_at: DateTime<Utc>,

    /// 更新时间
    pub updated_at: DateTime<Utc>,
}

/// 模型类型枚举
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "lowercase")]
pub enum ModelType {
    Image,
    Chat,
}

impl ModelConfig {
    /// 创建新的模型配置
    pub fn new(
        model_id: String,
        model_name: String,
        model_type: ModelType,
        base_url: String,
        api_key: String,
        api_url: Option<String>,
    ) -> Self {
        let now = Utc::now();
        Self {
            model_id,
            model_name,
            model_type,
            base_url,
            api_key,
            api_url,
            enabled: true,
            created_at: now,
            updated_at: now,
        }
    }

    /// 更新配置
    pub fn update(
        &mut self,
        model_name: Option<String>,
        base_url: Option<String>,
        api_key: Option<String>,
        api_url: Option<String>,
        enabled: Option<bool>,
    ) {
        if let Some(name) = model_name {
            self.model_name = name;
        }
        if let Some(url) = base_url {
            self.base_url = url;
        }
        if let Some(key) = api_key {
            self.api_key = key;
        }
        if let Some(url) = api_url {
            self.api_url = Some(url);
        }
        if let Some(en) = enabled {
            self.enabled = en;
        }
        self.updated_at = Utc::now();
    }
}

impl std::fmt::Display for ModelType {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        match self {
            ModelType::Image => write!(f, "image"),
            ModelType::Chat => write!(f, "chat"),
        }
    }
}

impl std::str::FromStr for ModelType {
    type Err = String;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        match s.to_lowercase().as_str() {
            "image" => Ok(ModelType::Image),
            "chat" => Ok(ModelType::Chat),
            _ => Err(format!("Invalid model type: {}", s)),
        }
    }
}
